### ISSUE TITLE
Duel Gamemode with queue system (like quake)

### ISSUE THREAD

**Issue by gustavomassa** [YaLTeR/OpenAG#167]
Based on the quake duel system, how hard would it be to implement a new game mode for duels? We already have a queue system for the arena game mode, we also have some special code for the CTF mode.

Where server-side files are located, is there any doc/wiki about the architecture of the HL-SDK/AG? I really can help with the code, but the project is old and huge, so I have no idea where to begin.
I found the CDiscArena class (disc_arena.h/cpp) from ricochet/dlls. Does AG uses it for Arena game mode?
I also found some arena code on the client files, most about hud and vgui files

We have the game modes folder inside the AG folder containing all the available game modes. I was able to find the server "mp" command codes, but I did not find any "sv_ag" command code. How and where these "sv_ag" commands are coded/handled?

**Comment by YaLTeR** [comment]
Yeah, this sounds like a largely server-side task. OpenAG contains only the client-side part, you can find the server repo here: https://github.com/fireblizzard/agmod
